tiledb/sm/array/array.cc

```diff
@@ -211,12 +211,31 @@ Status Array::open(
   RETURN_NOT_OK(
       encryption_key_.set_key(encryption_type, encryption_key, key_length));
 
-  timestamp_ =
-      (query_type == QueryType::READ) ? utils::time::timestamp_now_ms() : 0;
   metadata_.clear();
   metadata_loaded_ = false;
   non_empty_domain_computed_ = false;
 
+  bool found = false;
+  uint64_t timestamp_cfg = 0;
+  RETURN_NOT_OK(config_.get<uint64_t>(
+      "sm.array.open_timestamp_start", &timestamp_cfg, &found));
```

Comment:
There is a confusion here. Let's consider today's behavior. There may be fragment timestamps spanning from 0 to infinity (`UINT64_MAX`). When we open the array without a timestamp, we need to get all fragments. Therefore, the timestamp ranges is `[0, UINT64_MAX]`. When we open the array at a particular timestamp with current `open_at`, then we are implementing an **as of** behavior. That is, we need to view the array as of at the given timestamp, ignoring all the updates that happened after that. Therefore, that is equivalent to constraining the timestamp range to `[0, timestamp_end]`. 

So, in this entire PR, you effectively need to set and get the "end timestamp" to mimic today's behavior. Now there are a lot of cases where the user wants to see the most recent updates. In that case the user would set "timestamp start" to constrain the range to `[timestamp_start, UINT64_MAX]`. Finally, in some cases the user will want any arbitrary interval, i.e., constrain the range to `[timestamp_start, timestamp_end]`.